Fixes in execl call and dma's own reallocf implementation

colazivi opened the pull request:
In order to avoid problems on architectures that do not define NULL as ((void *) 0), the last parameter to execl should be explicitly cast to (char *).
See also man execl(3).

Also prevent a double free() in own reallocf() implementation. See also [FreeBSD's reallocf implementation](https://github.com/freebsd/freebsd/blob/master/lib/libc/stdlib/reallocf.c).

corecode commented on local.c:
```diff
@@ -82,7 +82,7 @@ create_mbox(const char *name)
 			close(i);
 
-		execl(LIBEXEC_PATH "/dma-mbox-create", "dma-mbox-create", name, NULL);
+		execl(LIBEXEC_PATH "/dma-mbox-create", "dma-mbox-create", name, (char *) NULL);
```
No space between `)` and `NULL`

colazivi replied:
Done.

colazivi commented:
I've also fixed a couple of compiler warnings primarily on Dragonfly BSD where the -Wextra flag is enabled by default:
Fallthrough comments didn't match [GCC's regexes](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-fallthrough), and added a missing function declaration.
Also resolved #83.

corecode commented:
Turns out that at least one missing fallthrough comment was a program logic bug;  the lesson here is to not just silence the compiler but to really understand whether the code should fall through or whether there is a hidden bug.  Adding the fallthrough comment hides it even more by implying intent.

colazivi commented:
I see, the one in mail.c was indeed a bug. No offense, but the compiler warning must have been ignored for years, so hiding it surely didn't help but my personal lesson here is: Don't rely on implicit fallthroughs. Especially not in a function that is already blessed with nested switch statements in a loop mixed with gotos ;-)

corecode commented:
I never saw a compiler warning there - maybe we need to update the compile flags?  Maybe this is a new warning?

corecode commented:
And I agree - this code is terrible; it should be ripped out and be replaced with a generated parser.

colazivi commented:
On a clean Dragonfly-BSD make's default CFLAGS include -Wextra and -Werror. -WExtra then enables -Wimplicit-fallthrough=3. I guess, that's when I saw this warning first. Probably -Wextra would be a good default setting for the whole project as well.

But, I certainly agree with getting rid of that code, that is simply unmaintainable in my eyes.
